File: Lua/gnome_sort.lua
function gnome_sort(arr)
    local index = 1
    while index <= #arr do
        if index == 1 or arr[index - 1] <= arr[index] then
            index = index + 1
        else
            arr[index], arr[index - 1] = arr[index - 1], arr[index]
            index = index - 1
        end
    end
end

local arr = {5, 1, 4, 2, 8}
gnome_sort(arr)
print(table.concat(arr, " "))

File: PHP/gnome_sort.php
<?php

function gnomeSort(array $arr): array {
    $index = 0;
    $n = count($arr);

    while ($index < $n) {
        if ($index == 0 || $arr[$index] >= $arr[$index - 1]) {
            $index++;
        } else {
            [$arr[$index], $arr[$index - 1]] = [$arr[$index - 1], $arr[$index]];
            $index--;
        }
    }

    return $arr;
}

$unsortedArray = [34, 2, 10, -9, 7, 0, 78, -66];
$sortedArray = gnomeSort($unsortedArray);

echo "Sorted Array: " . implode(", ", $sortedArray) . PHP_EOL;
?>
